modules/ballerina-core/src/main/java/org/ballerinalang/bre/BallerinaTransactionManager.java

```diff
@@ -58,12 +62,40 @@ public void setTransactionError(boolean transactionError) {
         this.transactionError = transactionError;
     }
 
-    public void beginTransactionBlock() {
+    public void beginTransactionBlock(int transactionID, int retryCount) {
+        allowedTransactionRetryCounts.put(transactionID, retryCount);
+        currentTransactionRetryCounts.put(transactionID, 0);
         ++transactionLevel;
     }
 
+    public int getAllowedRetryCount(int transactionId) {
+        Integer allowedRetryCount =  allowedTransactionRetryCounts.get(transactionId);
+        if (allowedRetryCount == null) {
```

Comment:
Removed with 4a135c5d98c366552a6e96c03a7b21a45db2c5e6 since transactionId should be there in the map. Thanks